controllers/cloudstackmachine_controller.go

```diff
@@ -167,7 +169,13 @@ func (r *CloudStackMachineReconciler) reconcile(
 	if !ok {
 		return ctrl.Result{}, errors.New("Bootstrap secret data not ok.")
 	}
-	userData := base64.StdEncoding.EncodeToString(value)
+
+	buf := &bytes.Buffer{}
```

Comment:
This all seems reasonable and if you say it works, then I'm good with that.

My only thought is can we turn this into a function called compress data or something similar? I like using methods to abstract these details such that the overall code flow is easier to follow.

Reply:
Extracted a function to the cloud package.